pkg/monitor/lease.go

```diff
@@ -0,0 +1,132 @@
+package monitor
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/milosgajdos/tenus"
+	"github.com/openshift/baremetal-runtimecfg/pkg/config"
+	"github.com/openshift/baremetal-runtimecfg/pkg/utils"
+	"github.com/sirupsen/logrus"
+)
+
+const monitorConf = "monitor.conf"
+const macPrefixQumranet = "00:1A:4A"
+const leaseInterfaceTemplate = "mc-%s"
+const leaseFile = "/tmp/my.lease"
+
+type VIP struct {
+	name string
+	ip   net.IP
+}
+
+func needLease(cfgPath string) (bool, error) {
+	monitorConfFile := filepath.Join(filepath.Dir(cfgPath), monitorConf)
+
+	_, err := os.Stat(monitorConfFile)
+
+	if err == nil {
+		return true, nil
+	}
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+func leaseVIPs(clusterName string, vips []VIP) error {
+	for _, vip := range vips {
+		vipFullName := fmt.Sprintf("%s-%s", clusterName, vip.name)
```

Comment:
what will be vipFullName? is there any length limit?

Reply:
Good point. Fixed
```golang
const interfaceMaxSize = 12

		vipFullName := fmt.Sprintf("%s-%s", clusterName, vip.name)
		// Takes the last `interfaceMaxSize` bytes of vipFullName
		vipFullName = vipFullName[int(math.Max(0, float64(len(vipFullName)-interfaceMaxSize))):]
```